polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaInterpreter.scala

```diff
@@ -76,6 +87,67 @@ class ScalaInterpreter(
     }
   }
 
+  private def importCellInstance(sym: global.ModuleSymbol) = {
+    val symType = global.exitingTyper(sym.toType)
+    val instanceSym = global.exitingTyper(symType.decl(global.TermName("INSTANCE")))
+    val instanceTyp = global.exitingTyper(instanceSym.info.finalResultType)
+
+    kernelContext.runInterruptibleIO(importToRuntime.importSymbol(sym)).flatMap {
+      runtimeSym =>
+        kernelContext.runInterruptibleIO {
+          try {
+            val moduleMirror = runtimeMirror.reflectModule(runtimeSym.asModule)
+            val moduleInst = runtimeMirror.reflect(moduleMirror.instance)
+            val runtimeInstanceSym = moduleInst.symbol.info.decl(runtime.universe.TermName("INSTANCE")).asTerm.accessed.asTerm
+            val instanceFieldMirror = moduleInst.reflectField(runtimeInstanceSym)
+            val instMirror = runtimeMirror.reflect(instanceFieldMirror.get)
+            (instMirror, instanceTyp, runtimeSym)
+          } catch {
+            case err: ExceptionInInitializerError =>
+              val e = err
+              throw RuntimeError(err.getCause)
+            case err: Throwable =>
+              val e = err
+              throw RuntimeError(err)
+          }
+        }
+    }
+  }
+
+  // matches a definition that we'll publish to the symbol table as a value
+  private object Val {
+    def unapply(sym: global.Symbol): Option[global.Symbol] = {
+
+      // make sure it doesn't have any parameters and isn't overloaded, or an object/case class constructor, or has parameters
+      // (have to check that it isn't a module first, or asMethod will fail even though isMethod is true. Yeesh.)
+      val methodWithParams = sym.alternatives.collect {
+        case method if method.isMethod && (method.isOverloaded || method.isModuleOrModuleClass || method.asMethod.paramLists.flatten.nonEmpty) => method
+      }
+
+      if (methodWithParams.nonEmpty) {
+        None
+      } else if (sym.isVal) {
+        Some(sym)
+      } else if (sym.isAccessor && !sym.isModule) {
+        Some(sym)
+      } else if (sym.isGetter && !sym.isModuleOrModuleClass) {
+        Some(sym)
+      } else None
+    }
+  }
+
+  // matches a definition that we'll publish to the symbol table as a function (if possible)
+  private object Def {
+    def unapply(sym: global.Symbol): Option[global.MethodSymbol] = if (sym.isMethod) {
+      if (sym.isOverloaded || sym.isModuleOrModuleClass)
+        None
+      else if (sym.alternatives.exists(!_.isMethod) && sym.alternatives.exists(_.isMethod))
```

Comment:
What kind of symbol satisfies this condition?

Reply:
`case class Foo()` – since it's now an inner class, its companion object is both a method (of the outer class) and a module.

Reply:
I guess I should add comments there about what I'm looking for with these cases. It was very trial & error since the semantics of all of `Symbol`'s methods seem to be arcane knowledge (most of which I don't have)